rest/taskana-rest-spring/src/main/java/pro/taskana/rest/MonitorController.java

```diff
@@ -30,53 +33,66 @@
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Basket1\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Basket2\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Basket3\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Basket4\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Gruppenpostkorb KSC\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Gruppenpostkorb KSC 1\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Gruppenpostkorb KSC 2\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"PPK Teamlead KSC 1\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"PPK Teamlead KSC 2\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"PPK User 1 KSC 1\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"PPK User 2 KSC 1\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"PPK User 1 KSC 2\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"PPK User 2 KSC 2\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Gruppenpostkorb KSC B\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Gruppenpostkorb KSC B1\"},");
-        builder.append("{\"data\": [0,0,0,0,0,0,0,0,0,0,0],\"label\": \"Gruppenpostkorb KSC B2\"}");
-        builder.append("]}");
-        return ResponseEntity.status(HttpStatus.OK).body(builder.toString());
     }
 
-    @GetMapping(path = "/taskStatusReport")
+    @GetMapping(path = "/tasks-classification-report")
     @Transactional(readOnly = true, rollbackFor = Exception.class)
-    public ResponseEntity<ReportResource> getTaskStatusReport(@RequestParam(required = false) List<String> domains,
-        @RequestParam(required = false) List<TaskState> states) throws NotAuthorizedException {
-        // return ResponseEntity.status(HttpStatus.OK)
-        // .body(reportAssembler.toResource(taskMonitorService.getTaskStatusReport(domains, states), domains, states));
+    public ResponseEntity<ReportResource> getTasksClassificationReport()
+        throws NotAuthorizedException, InvalidArgumentException {
+
         return ResponseEntity.status(HttpStatus.OK)
             .body(reportAssembler.toResource(
-                taskMonitorService.createTaskStatusReportBuilder().stateIn(states).domainIn(domains).buildReport(),
-                domains, states));
+                taskMonitorService.createClassificationReportBuilder()
+                    .withColumnHeaders(getTaskClassificationTimeInterval())
+                    .buildReport()));
+    }
+
+    private List<TimeIntervalColumnHeader> getTaskClassificationTimeInterval() {
+        List<TimeIntervalColumnHeader> columnHeaders = new ArrayList<>();
```

Comment:
just for you (don't have to implement this). The stream API is real nice for those operations.

return Stream.concat(
            Stream.of(new TimeIntervalColumnHeader(Integer.MIN_VALUE, -10),
                new TimeIntervalColumnHeader(-10, -5),
                new TimeIntervalColumnHeader(5, 10),
                new TimeIntervalColumnHeader(10, Integer.MAX_VALUE)),
            Stream.of(-4, -3, -2, -1, 0, 1, 2, 3, 4)
                .map(TimeIntervalColumnHeader::new))
            .collect(Collectors.toList());

Reply:
Sure I will do it, it's shorter than mine.